docs/database-structure.md

```diff
@@ -0,0 +1,336 @@
+- attrs: See [object attributes](attributes.md).
+- childrenAttrs: Attributes that are applied to all the children objects.
+- createdAt
+- updatedAt
+- externalRelationUrl (Maybe not needed).
+- authorId (Maybe not needed).
+- campus: The campus the object is related to.
+- referenceId: A reference to the base object.
+- referencedAt: The timestamp of the creation of the copy object.
+
+### object_child
+
+It keeps a child-parent relationship between two objects. It's used to encapsulate objects inside each other.
+
+Columns:
+- id
+- parentId
+- childId
+- attrs
+- key: When generating the JS object this field will be the `key` of the child.
+- index: Defines the position of the child object inside the parent object.
+
+### object_status
+
+Columns:
+- status
+Possible values:
+  - `online`
+  - `offline`
+  - `draft`
+
+### object_type
+
+Columns:
+- type
+Possible values:
+  - `onboarding`
+  - `campus`
+  - `exercise`
+  - `quest`
+  - `signup`
+  - `exam`
+  - `raid`
+  - `project`
+  - `piscine`
+
+### progress
+
+It is a register of the steps in the progression of students doing the exercises and projects.
+
+Columns:
+- id
+- createdAt
+- updateAt
+- userId
+- groupId
+- eventId
+- version
+- grade
+- isDone
+- path
+- campus
+- objectId
+
+### record
+
+Keeps a register for the bans. It can also keep the record of other situations, this is done by setting the banEndAt to a date already passed and the reason for the record.
+
+Columns:
+- id
+- userId
+- authorId
+- message
+- banEndAt
+- createdAt
+
+### registration
+
+Registrations to events (For example, an exam).
+
+Columns:
+- id
+- createdAt, When the registration was created.
+- startAt, When users can start registering to an event.
+- endAt, When the registration ends.
+- eventStartAt, When the event the registration refers to starts.
+- objectId, The object the registration refers to.
+- parentId, The parent object of the object the registration refers to.
+- attrs
+- path
+- campus
+
+### registration_user
+
+The user registered to events. A user can register to many events and and event can have many users.
+
+Columns:
+- id
+- registrationId
+- userId
+- createdAt
+
+### result
+
+The result for each exercise and project made in the platform.
+
+Columns:
+- id
+- createdAt
+- updatedAt
+- grade
+- progressId
+- attrs
+- type, For the possible types see the table [result_type](#result_type)
+- userId
+- groupId
+- objectId
+- path
+- version, The version of the code the result refers to.
+- eventId
+- isLast, Used to represent transitory states. A progress is only considered finished when `isLast` is `true`.
+- campus
+
+### result_type
+
+Columns:
+- type
+  - `tester`
+  - `user_audit`
+  - `admit_audit`
+  - `admin_selection`
+  - `status`
+
+### role
+
+Columns:
+- id
+- slug
+- name
+- description
+- createdAt
+- updatedAt
+
+### token
+
+Columns:
+- id
+- status
+- createdAt
+- updatedAt
+
+### token_status
+
+Columns:
+- status
+  - `active`
+  - `expired`
```

Comment:
I think it would make sense to skip definitions and inline enum tables values where they are used:
```md
### token

Columns:
- id
- status (enum)
  - `active`
  - `expired`
- createdAt
- updatedAt
```

the data is design like this per requirement by hasura https://hasura.io/docs/latest/graphql/core/databases/postgres/schema/enums.html